src/app/SharpRaven/Data/HttpRequester.cs

```diff
@@ -139,21 +120,26 @@ public RequestData Data
         }
 
         /// <summary>
-        /// Gets the <see cref="JsonPacket"/> being sent to Sentry.
+        /// Gets the <see cref="HttpWebRequest"/> instance being used to perform the HTTP request to Sentry.
         /// </summary>
-        public JsonPacket Packet
```

Comment:
True. I couldn't though understand what was the goal of exposing `JsonPacket` from `Requester` considering that any modification to it would not reflect to the request sent.

The request [will read `data` upon invoked](https://github.com/getsentry/raven-csharp/pull/242/files#diff-3dfc9454bda7408e3ff2f2c575c2fdb3L168).

So one has a reference to the `RavenClient` and `JsonPacket` and with that, creates a `Requester`. Why pass both references into it and expose publicly via properties?

Perhaps I'm missing the big picture though but my goal was to try decoupling those types a bit by removing dependency of `RavenClient` and `JsonPacket` from the `HttpRequester` and depend only `RequestData`.

Reply:
I don't quite understand it either. Not sure why it was designed like that and I can't remember what I thought when that was done either. 😄 Anyway, I welcome the proposed changes, just beware it's breaking backwards compatibility, warranting a new major version.

Reply:
Sounds great! There's a few more things to fix (reporting the architecture, parsing runtime on the backend and this change) but once those things are done and we can prepare a major release then.